sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -791,6 +792,22 @@ abstract class TypeChecks[RET] {
       meta.willNotWorkOnGpu(msgFormat.format(stringifyTypeAttributeMap(unsupportedTypes)))
     }
   }
+
+  /** Add more infomation in log when timezone not UTC */
+  protected def tagTimestampUtcCheck(
+    meta: RapidsMeta[_, _, _],
+    fields: Seq[StructField]
+    ): Unit = {
+    val schemaHasTimestamps = fields.exists { field =>
+      TrampolineUtil.dataTypeExistsRecursively(field.dataType, _.isInstanceOf[TimestampType])
+    }
+    if (schemaHasTimestamps && !(TypeChecks.areTimestampsSupported(ZoneId.systemDefault()) &&
```

Comment:
Duplicated, can we extract a `util` method?

https://github.com/NVIDIA/spark-rapids/blob/branch-22.10/sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala#L423toL429